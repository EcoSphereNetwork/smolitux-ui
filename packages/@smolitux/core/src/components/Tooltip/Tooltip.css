.smolitux-tooltip-container {
  position: relative;
  display: inline-block;
}

.smolitux-tooltip {
  position: absolute;
  z-index: 1000;
  padding: 0.5rem 0.75rem;
  font-size: 0.875rem;
  line-height: 1.25rem;
  color: var(--tooltip-text-color, white);
  background-color: var(--tooltip-bg, #1f2937);
  border-radius: 0.375rem;
  box-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.1), 0 2px 4px -1px rgba(0, 0, 0, 0.06);
  max-width: 250px;
  width: max-content;
  pointer-events: none;
  opacity: 0;
  visibility: hidden;
  transition: opacity 0.15s ease-in-out, visibility 0.15s ease-in-out;
}

.smolitux-tooltip--visible {
  opacity: 1;
  visibility: visible;
}

/* Tooltip-Pfeil */
.smolitux-tooltip-arrow {
  position: absolute;
  width: 0;
  height: 0;
  border-style: solid;
}

/* Pfeil-Positionen */
.smolitux-tooltip--top .smolitux-tooltip-arrow {
  bottom: -0.25rem;
  border-width: 0.25rem 0.25rem 0 0.25rem;
  border-color: var(--tooltip-bg, #1f2937) transparent transparent transparent;
}

.smolitux-tooltip--right .smolitux-tooltip-arrow {
  left: -0.25rem;
  border-width: 0.25rem 0.25rem 0.25rem 0;
  border-color: transparent var(--tooltip-bg, #1f2937) transparent transparent;
}

.smolitux-tooltip--bottom .smolitux-tooltip-arrow {
  top: -0.25rem;
  border-width: 0 0.25rem 0.25rem 0.25rem;
  border-color: transparent transparent var(--tooltip-bg, #1f2937) transparent;
}

.smolitux-tooltip--left .smolitux-tooltip-arrow {
  right: -0.25rem;
  border-width: 0.25rem 0 0.25rem 0.25rem;
  border-color: transparent transparent transparent var(--tooltip-bg, #1f2937);
}

/* Tooltip-Inhalt */
.smolitux-tooltip-content {
  position: relative;
  z-index: 1;
}

/* Animationen */
@keyframes smolitux-tooltip-fade-in {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

@keyframes smolitux-tooltip-fade-out {
  from {
    opacity: 1;
  }
  to {
    opacity: 0;
  }
}

.smolitux-tooltip--animate-in {
  animation: smolitux-tooltip-fade-in 0.15s ease-in-out forwards;
}

.smolitux-tooltip--animate-out {
  animation: smolitux-tooltip-fade-out 0.1s ease-in-out forwards;
}

/* Positionsvarianten mit Offset */
.smolitux-tooltip--top {
  bottom: calc(100% + 0.5rem);
  left: 50%;
  transform: translateX(-50%);
}

.smolitux-tooltip--right {
  left: calc(100% + 0.5rem);
  top: 50%;
  transform: translateY(-50%);
}

.smolitux-tooltip--bottom {
  top: calc(100% + 0.5rem);
  left: 50%;
  transform: translateX(-50%);
}

.smolitux-tooltip--left {
  right: calc(100% + 0.5rem);
  top: 50%;
  transform: translateY(-50%);
}

/* Varianten */
.smolitux-tooltip--info {
  background-color: var(--tooltip-info-bg, #3b82f6);
}

.smolitux-tooltip--info .smolitux-tooltip-arrow {
  border-color: var(--tooltip-info-bg, #3b82f6);
}

.smolitux-tooltip--info.smolitux-tooltip--top .smolitux-tooltip-arrow {
  border-color: var(--tooltip-info-bg, #3b82f6) transparent transparent transparent;
}

.smolitux-tooltip--info.smolitux-tooltip--right .smolitux-tooltip-arrow {
  border-color: transparent var(--tooltip-info-bg, #3b82f6) transparent transparent;
}

.smolitux-tooltip--info.smolitux-tooltip--bottom .smolitux-tooltip-arrow {
  border-color: transparent transparent var(--tooltip-info-bg, #3b82f6) transparent;
}

.smolitux-tooltip--info.smolitux-tooltip--left .smolitux-tooltip-arrow {
  border-color: transparent transparent transparent var(--tooltip-info-bg, #3b82f6);
}

.smolitux-tooltip--success {
  background-color: var(--tooltip-success-bg, #10b981);
}

.smolitux-tooltip--success.smolitux-tooltip--top .smolitux-tooltip-arrow {
  border-color: var(--tooltip-success-bg, #10b981) transparent transparent transparent;
}

.smolitux-tooltip--success.smolitux-tooltip--right .smolitux-tooltip-arrow {
  border-color: transparent var(--tooltip-success-bg, #10b981) transparent transparent;
}

.smolitux-tooltip--success.smolitux-tooltip--bottom .smolitux-tooltip-arrow {
  border-color: transparent transparent var(--tooltip-success-bg, #10b981) transparent;
}

.smolitux-tooltip--success.smolitux-tooltip--left .smolitux-tooltip-arrow {
  border-color: transparent transparent transparent var(--tooltip-success-bg, #10b981);
}

.smolitux-tooltip--warning {
  background-color: var(--tooltip-warning-bg, #f59e0b);
}

.smolitux-tooltip--warning.smolitux-tooltip--top .smolitux-tooltip-arrow {
  border-color: var(--tooltip-warning-bg, #f59e0b) transparent transparent transparent;
}

.smolitux-tooltip--warning.smolitux-tooltip--right .smolitux-tooltip-arrow {
  border-color: transparent var(--tooltip-warning-bg, #f59e0b) transparent transparent;
}

.smolitux-tooltip--warning.smolitux-tooltip--bottom .smolitux-tooltip-arrow {
  border-color: transparent transparent var(--tooltip-warning-bg, #f59e0b) transparent;
}

.smolitux-tooltip--warning.smolitux-tooltip--left .smolitux-tooltip-arrow {
  border-color: transparent transparent transparent var(--tooltip-warning-bg, #f59e0b);
}

.smolitux-tooltip--danger {
  background-color: var(--tooltip-danger-bg, #ef4444);
}

.smolitux-tooltip--danger.smolitux-tooltip--top .smolitux-tooltip-arrow {
  border-color: var(--tooltip-danger-bg, #ef4444) transparent transparent transparent;
}

.smolitux-tooltip--danger.smolitux-tooltip--right .smolitux-tooltip-arrow {
  border-color: transparent var(--tooltip-danger-bg, #ef4444) transparent transparent;
}

.smolitux-tooltip--danger.smolitux-tooltip--bottom .smolitux-tooltip-arrow {
  border-color: transparent transparent var(--tooltip-danger-bg, #ef4444) transparent;
}

.smolitux-tooltip--danger.smolitux-tooltip--left .smolitux-tooltip-arrow {
  border-color: transparent transparent transparent var(--tooltip-danger-bg, #ef4444);
}

/* Größen */
.smolitux-tooltip--sm {
  padding: 0.25rem 0.5rem;
  font-size: 0.75rem;
}

.smolitux-tooltip--lg {
  padding: 0.75rem 1rem;
  font-size: 1rem;
}

/* Dark Mode */
.dark .smolitux-tooltip {
  color: var(--tooltip-text-color-dark, #f9fafb);
  background-color: var(--tooltip-bg-dark, #374151);
}

.dark .smolitux-tooltip--top .smolitux-tooltip-arrow {
  border-color: var(--tooltip-bg-dark, #374151) transparent transparent transparent;
}

.dark .smolitux-tooltip--right .smolitux-tooltip-arrow {
  border-color: transparent var(--tooltip-bg-dark, #374151) transparent transparent;
}

.dark .smolitux-tooltip--bottom .smolitux-tooltip-arrow {
  border-color: transparent transparent var(--tooltip-bg-dark, #374151) transparent;
}

.dark .smolitux-tooltip--left .smolitux-tooltip-arrow {
  border-color: transparent transparent transparent var(--tooltip-bg-dark, #374151);
}