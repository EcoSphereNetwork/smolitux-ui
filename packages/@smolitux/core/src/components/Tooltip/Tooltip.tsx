import React, { useState, useRef, useEffect } from 'react';

export interface TooltipProps {
  /** Tooltip-Inhalt */
  content: React.ReactNode;
  /** Das Element, das mit dem Tooltip versehen werden soll */
  children: React.ReactElement;
  /** Position des Tooltips */
  position?: 'top' | 'right' | 'bottom' | 'left';
  /** Verzögerung vor dem Anzeigen (in ms) */
  delay?: number;
  /** Verzögerung vor dem Verstecken (in ms) */
  hideDelay?: number;
  /** Maximale Breite des Tooltips */
  maxWidth?: number | string;
  /** Zusätzliche CSS-Klassen */
  className?: string;
  /** Deaktiviert den Tooltip */
  disabled?: boolean;
  /** Pfeil anzeigen */
  arrow?: boolean;
}

/**
 * Tooltip-Komponente für Hilfetexte
 */
export const Tooltip: React.FC<TooltipProps> = ({
  content,
  children,
  position = 'top',
  delay = 200,
  hideDelay = 100,
  maxWidth = 250,
  className = '',
  disabled = false,
  arrow = true,
}) => {
  const [isVisible, setIsVisible] = useState(false);
  const [tooltipPosition, setTooltipPosition] = useState({ top: 0, left: 0 });
  const [arrowPosition, setArrowPosition] = useState({ top: 0, left: 0 });
  const triggerRef = useRef<HTMLElement>(null);
  const tooltipRef = useRef<HTMLDivElement>(null);
  const showTimeoutRef = useRef<NodeJS.Timeout | null>(null);
  const hideTimeoutRef = useRef<NodeJS.Timeout | null>(null);
  
  // Positionierung des Tooltips und Pfeils berechnen
  const calculatePosition = () => {
    if (!triggerRef.current || !tooltipRef.current) return;
    
    const triggerRect = triggerRef.current.getBoundingClientRect();
    const tooltipRect = tooltipRef.current.getBoundingClientRect();
    const scrollTop = window.pageYOffset || document.documentElement.scrollTop;
    const scrollLeft = window.pageXOffset || document.documentElement.scrollLeft;
    
    // Basisposition basierend auf der gewählten Position
    let top = 0;
    let left = 0;
    
    // Pfeilposition
    let arrowTop = 0;
    let arrowLeft = 0;
    
    switch (position) {
      case 'top':
        top = triggerRect.top + scrollTop - tooltipRect.height - 10;
        left = triggerRect.left + scrollLeft + (triggerRect.width / 2) - (tooltipRect.width / 2);
        arrowTop = tooltipRect.height;
        arrowLeft = tooltipRect.width / 2;
        break;
      case 'right':
        top = triggerRect.top + scrollTop + (triggerRect.height / 2) - (tooltipRect.height / 2);
        left = triggerRect.right + scrollLeft + 10;
        arrowTop = tooltipRect.height / 2;
        arrowLeft = -5;
        break;
      case 'bottom':
        top = triggerRect.bottom + scrollTop + 10;
        left = triggerRect.left + scrollLeft + (triggerRect.width / 2) - (tooltipRect.width / 2);
        arrowTop = -5;
        arrowLeft = tooltipRect.width / 2;
        break;
      case 'left':
        top = triggerRect.top + scrollTop + (triggerRect.height / 2) - (tooltipRect.height / 2);
        left = triggerRect.left + scrollLeft - tooltipRect.width - 10;
        arrowTop = tooltipRect.height / 2;
        arrowLeft = tooltipRect.width;
        break;
    }
    
    // Stellen Sie sicher, dass der Tooltip im Viewport bleibt
    const viewportWidth = window.innerWidth;
    const viewportHeight = window.innerHeight;
    
    // Horizontal anpassen
    if (left < 0) {
      arrowLeft = arrowLeft + left; // Pfeil entsprechend verschieben
      left = 10;
    } else if (left + tooltipRect.width > viewportWidth) {
      const overflow = left + tooltipRect.width - viewportWidth;
      arrowLeft = arrowLeft + overflow; // Pfeil entsprechend verschieben
      left = viewportWidth - tooltipRect.width - 10;
    }
    
    // Vertikal anpassen
    if (top < 0) {
      arrowTop = arrowTop + top; // Pfeil entsprechend verschieben
      top = 10;
    } else if (top + tooltipRect.height > viewportHeight + scrollTop) {
      const overflow = top + tooltipRect.height - (viewportHeight + scrollTop);
      arrowTop = arrowTop + overflow; // Pfeil entsprechend verschieben
      top = viewportHeight + scrollTop - tooltipRect.height - 10;
    }
    
    setTooltipPosition({ top, left });
    setArrowPosition({ top: arrowTop, left: arrowLeft });
  };
  
  // Show/Hide Logik mit Verzögerung
  const showTooltip = () => {
    if (disabled) return;
    
    // Clear any existing timers
    if (hideTimeoutRef.current) {
      clearTimeout(hideTimeoutRef.current);
      hideTimeoutRef.current = null;
    }
    
    // Set show timer
    showTimeoutRef.current = setTimeout(() => {
      setIsVisible(true);
      // Erst berechnen, wenn der Tooltip sichtbar ist
      setTimeout(calculatePosition, 0);
    }, delay);
  };
  
  const hideTooltip = () => {
    if (showTimeoutRef.current) {
      clearTimeout(showTimeoutRef.current);
      showTimeoutRef.current = null;
    }
    
    hideTimeoutRef.current = setTimeout(() => {
      setIsVisible(false);
    }, hideDelay);
  };
  
  // Clean up timeouts on unmount
  useEffect(() => {
    return () => {
      if (showTimeoutRef.current) clearTimeout(showTimeoutRef.current);
      if (hideTimeoutRef.current) clearTimeout(hideTimeoutRef.current);
    };
  }, []);
  
  // Recalculate on window resize
  useEffect(() => {
    if (!isVisible) return;
    
    const handleResize = () => calculatePosition();
    window.addEventListener('resize', handleResize);
    
    return () => {
      window.removeEventListener('resize', handleResize);
    };
  }, [isVisible]);
  
  // Pfeilrenderrichtung
  const getArrowStyle = () => {
    switch (position) {
      case 'top':
        return {
          borderTop: '5px solid currentColor',
          borderLeft: '5px solid transparent',
          borderRight: '5px solid transparent',
          bottom: '-5px'
        };
      case 'right':
        return {
          borderRight: '5px solid currentColor',
          borderTop: '5px solid transparent',
          borderBottom: '5px solid transparent',
          left: '-5px'
        };
      case 'bottom':
        return {
          borderBottom: '5px solid currentColor',
          borderLeft: '5px solid transparent',
          borderRight: '5px solid transparent',
          top: '-5px'
        };
      case 'left':
        return {
          borderLeft: '5px solid currentColor',
          borderTop: '5px solid transparent',
          borderBottom: '5px solid transparent',
          right: '-5px'
        };
    }
  };
  
  // Trigger-Element mit Event Listenern klonen
  const triggerElement = React.cloneElement(React.Children.only(children), {
    ref: (element: HTMLElement | null) => {
      // Ref vom Original beibehalten, wenn vorhanden
      const child = React.Children.only(children);
      if (React.isValidElement(child) && 'ref' in child) {
        const childRef = (child as any).ref;
        if (typeof childRef === 'function') {
          childRef(element);
        } else if (childRef && typeof childRef === 'object' && 'current' in childRef) {
          (childRef as React.MutableRefObject<HTMLElement | null>).current = element;
        }
      }
      if (triggerRef) {
        (triggerRef as React.MutableRefObject<HTMLElement | null>).current = element;
      }
    },
    onMouseEnter: (e: React.MouseEvent<HTMLElement>) => {
      showTooltip();
      // Original-Handler beibehalten
      if (children.props.onMouseEnter) {
        children.props.onMouseEnter(e);
      }
    },
    onMouseLeave: (e: React.MouseEvent<HTMLElement>) => {
      hideTooltip();
      // Original-Handler beibehalten
      if (children.props.onMouseLeave) {
        children.props.onMouseLeave(e);
      }
    },
    onFocus: (e: React.FocusEvent<HTMLElement>) => {
      showTooltip();
      // Original-Handler beibehalten
      if (children.props.onFocus) {
        children.props.onFocus(e);
      }
    },
    onBlur: (e: React.FocusEvent<HTMLElement>) => {
      hideTooltip();
      // Original-Handler beibehalten
      if (children.props.onBlur) {
        children.props.onBlur(e);
      }
    }
  });
  
  return (
    <>
      {triggerElement}
      
      {isVisible && (
        <div
          ref={tooltipRef}
          className={`
            fixed z-50 p-2 text-sm text-white rounded shadow-md
            bg-gray-800 dark:bg-gray-700
            tooltip tooltip-${position} tooltip-content
            ${className}
          `}
          style={{ 
            top: tooltipPosition.top,
            left: tooltipPosition.left,
            maxWidth: typeof maxWidth === 'number' ? `${maxWidth}px` : maxWidth
          }}
          role="tooltip"
        >
          {content}
          
          {arrow && (
            <div
              data-testid="tooltip-arrow"
              className="absolute w-0 h-0 text-gray-800 dark:text-gray-700"
              style={{
                ...getArrowStyle(),
                top: arrowPosition.top === 0 ? getArrowStyle().top : arrowPosition.top,
                left: arrowPosition.left === 0 ? getArrowStyle().left : arrowPosition.left,
                right: getArrowStyle().right
              }}
            ></div>
          )}
        </div>
      )}
    </>
  );
};

export default Tooltip;
