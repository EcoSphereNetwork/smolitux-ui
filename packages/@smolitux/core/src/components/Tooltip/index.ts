export { Tooltip } from './Tooltip';
export type { TooltipProps } from './Tooltip';

// Fur Abwartskompatibilitat mit dem bestehenden Export
export { Tooltip as default } from './Tooltip';