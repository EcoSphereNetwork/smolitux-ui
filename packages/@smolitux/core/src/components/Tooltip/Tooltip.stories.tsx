import React from 'react';
import { Meta, StoryObj } from '@storybook/react';
import { Tooltip } from './Tooltip';
import { Button } from '../Button/Button';

const meta: Meta<typeof Tooltip> = {
  title: 'Core/Tooltip',
  component: Tooltip,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    placement: {
      control: { type: 'select' },
      options: [
        'top', 'top-start', 'top-end',
        'right', 'right-start', 'right-end',
        'bottom', 'bottom-start', 'bottom-end',
        'left', 'left-start', 'left-end'
      ],
    },
    delay: { control: { type: 'number' } },
    hideDelay: { control: { type: 'number' } },
    maxWidth: { control: { type: 'text' } },
    disabled: { control: { type: 'boolean' } },
    arrow: { control: { type: 'boolean' } },
    isOpen: { control: { type: 'boolean' } },
    isToggleable: { control: { type: 'boolean' } },
    closeOnOutsideClick: { control: { type: 'boolean' } },
    closeOnEsc: { control: { type: 'boolean' } },
    closeOnScroll: { control: { type: 'boolean' } },
    closeOnTooltipClick: { control: { type: 'boolean' } },
    offset: { control: { type: 'number' } },
    backgroundColor: { control: { type: 'color' } },
    textColor: { control: { type: 'color' } },
    animated: { control: { type: 'boolean' } },
    shadow: { control: { type: 'boolean' } },
    rounded: { control: { type: 'boolean' } },
    bordered: { control: { type: 'boolean' } },
    borderColor: { control: { type: 'color' } },
  },
};

export default meta;
type Story = StoryObj<typeof Tooltip>;

export const Default: Story = {
  args: {
    content: 'Dies ist ein Tooltip mit Hilfeinformationen',
    placement: 'top',
    delay: 200,
    hideDelay: 100,
    arrow: true,
    animated: true,
    shadow: true,
    rounded: true,
    children: <Button>Hover über mich</Button>,
  },
};

export const Placements: Story = {
  render: () => (
    <div className="grid grid-cols-3 gap-4">
      <Tooltip content="Top" placement="top">
        <Button className="w-full">Top</Button>
      </Tooltip>
      <Tooltip content="Top Start" placement="top-start">
        <Button className="w-full">Top Start</Button>
      </Tooltip>
      <Tooltip content="Top End" placement="top-end">
        <Button className="w-full">Top End</Button>
      </Tooltip>
      
      <Tooltip content="Right" placement="right">
        <Button className="w-full">Right</Button>
      </Tooltip>
      <Tooltip content="Right Start" placement="right-start">
        <Button className="w-full">Right Start</Button>
      </Tooltip>
      <Tooltip content="Right End" placement="right-end">
        <Button className="w-full">Right End</Button>
      </Tooltip>
      
      <Tooltip content="Bottom" placement="bottom">
        <Button className="w-full">Bottom</Button>
      </Tooltip>
      <Tooltip content="Bottom Start" placement="bottom-start">
        <Button className="w-full">Bottom Start</Button>
      </Tooltip>
      <Tooltip content="Bottom End" placement="bottom-end">
        <Button className="w-full">Bottom End</Button>
      </Tooltip>
      
      <Tooltip content="Left" placement="left">
        <Button className="w-full">Left</Button>
      </Tooltip>
      <Tooltip content="Left Start" placement="left-start">
        <Button className="w-full">Left Start</Button>
      </Tooltip>
      <Tooltip content="Left End" placement="left-end">
        <Button className="w-full">Left End</Button>
      </Tooltip>
    </div>
  ),
};

export const WithHTMLContent: Story = {
  args: {
    content: (
      <div>
        <h3 className="font-bold mb-2">Formatierter Tooltip</h3>
        <p>Tooltips können auch <strong>HTML-Inhalte</strong> enthalten.</p>
        <ul className="list-disc pl-4 mt-2">
          <li>Punkt 1</li>
          <li>Punkt 2</li>
          <li>Punkt 3</li>
        </ul>
      </div>
    ),
    maxWidth: 300,
    placement: 'right',
    children: <Button>Hover für komplexen Tooltip</Button>,
  },
};

export const Toggleable: Story = {
  args: {
    content: 'Klicken Sie erneut, um diesen Tooltip zu schließen',
    isToggleable: true,
    placement: 'top',
    children: <Button>Klicken für Tooltip</Button>,
  },
};

export const CustomStyling: Story = {
  args: {
    content: 'Benutzerdefinierter Tooltip mit eigenen Farben',
    backgroundColor: '#6366F1', // Indigo
    textColor: '#FFFFFF',
    borderColor: '#4F46E5',
    bordered: true,
    shadow: true,
    placement: 'top',
    children: <Button>Hover für benutzerdefinierten Tooltip</Button>,
  },
};

export const WithoutArrow: Story = {
  args: {
    content: 'Tooltip ohne Pfeil',
    arrow: false,
    placement: 'top',
    children: <Button>Hover über mich</Button>,
  },
};

export const WithDelay: Story = {
  args: {
    content: 'Dieser Tooltip erscheint mit Verzögerung',
    delay: 1000, // 1 Sekunde Verzögerung
    placement: 'top',
    children: <Button>Hover für verzögerten Tooltip</Button>,
  },
};

export const CloseOnClick: Story = {
  args: {
    content: 'Klicken Sie auf diesen Tooltip, um ihn zu schließen',
    closeOnTooltipClick: true,
    placement: 'top',
    children: <Button>Hover und dann auf Tooltip klicken</Button>,
  },
};