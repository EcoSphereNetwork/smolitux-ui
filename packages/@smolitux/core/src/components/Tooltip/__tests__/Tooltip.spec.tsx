import React from 'react';
import { render } from '@testing-library/react';
import { Tooltip } from '../Tooltip';

describe('Tooltip Snapshots', () => {
  it('renders tooltip correctly', () => {
    const { asFragment } = render(
      <Tooltip content="Tooltip Content">
        <button>Hover Me</button>
      </Tooltip>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders tooltip with different positions correctly', () => {
    const positions: Array<'top' | 'right' | 'bottom' | 'left'> = ['top', 'right', 'bottom', 'left'];
    
    const fragments = positions.map(position => {
      const { asFragment } = render(
        <Tooltip content="Tooltip Content" position={position}>
          <button>Hover Me</button>
        </Tooltip>
      );
      return { position, fragment: asFragment() };
    });
    
    fragments.forEach(({ position, fragment }) => {
      expect(fragment).toMatchSnapshot(`Tooltip with position ${position}`);
    });
  });

  it('renders tooltip with arrow correctly', () => {
    const { asFragment } = render(
      <Tooltip content="Tooltip Content" arrow>
        <button>Hover Me</button>
      </Tooltip>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders tooltip with custom maxWidth correctly', () => {
    const { asFragment } = render(
      <Tooltip content="Tooltip Content" maxWidth={200}>
        <button>Hover Me</button>
      </Tooltip>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders tooltip with custom className correctly', () => {
    const { asFragment } = render(
      <Tooltip content="Tooltip Content" className="custom-tooltip">
        <button>Hover Me</button>
      </Tooltip>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders tooltip with rich content correctly', () => {
    const { asFragment } = render(
      <Tooltip 
        content={
          <div>
            <h4>Tooltip Title</h4>
            <p>Tooltip description</p>
          </div>
        }
      >
        <button>Hover Me</button>
      </Tooltip>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders disabled tooltip correctly', () => {
    const { asFragment } = render(
      <Tooltip content="Tooltip Content" disabled>
        <button>Hover Me</button>
      </Tooltip>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});