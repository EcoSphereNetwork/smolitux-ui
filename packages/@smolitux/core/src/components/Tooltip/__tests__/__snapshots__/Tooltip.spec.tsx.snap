// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Snapshots renders disabled tooltip correctly 1`] = `
<DocumentFragment>
  <button>
    Hover Me
  </button>
</DocumentFragment>
`;

exports[`Tooltip Snapshots renders tooltip correctly 1`] = `
<DocumentFragment>
  <button>
    Hover Me
  </button>
</DocumentFragment>
`;

exports[`Tooltip Snapshots renders tooltip with arrow correctly 1`] = `
<DocumentFragment>
  <button>
    Hover Me
  </button>
</DocumentFragment>
`;

exports[`Tooltip Snapshots renders tooltip with custom className correctly 1`] = `
<DocumentFragment>
  <button>
    Hover Me
  </button>
</DocumentFragment>
`;

exports[`Tooltip Snapshots renders tooltip with custom maxWidth correctly 1`] = `
<DocumentFragment>
  <button>
    Hover Me
  </button>
</DocumentFragment>
`;

exports[`Tooltip Snapshots renders tooltip with different positions correctly: Tooltip with position bottom 1`] = `
<DocumentFragment>
  <button>
    Hover Me
  </button>
</DocumentFragment>
`;

exports[`Tooltip Snapshots renders tooltip with different positions correctly: Tooltip with position left 1`] = `
<DocumentFragment>
  <button>
    Hover Me
  </button>
</DocumentFragment>
`;

exports[`Tooltip Snapshots renders tooltip with different positions correctly: Tooltip with position right 1`] = `
<DocumentFragment>
  <button>
    Hover Me
  </button>
</DocumentFragment>
`;

exports[`Tooltip Snapshots renders tooltip with different positions correctly: Tooltip with position top 1`] = `
<DocumentFragment>
  <button>
    Hover Me
  </button>
</DocumentFragment>
`;

exports[`Tooltip Snapshots renders tooltip with rich content correctly 1`] = `
<DocumentFragment>
  <button>
    Hover Me
  </button>
</DocumentFragment>
`;
